### PR TITLE
Picard tools AddOrReplaceReadGroup

### PR DESCRIPTION
**Pull request by J35P312**
Hello there!
Here's my first try of adding a readgroup to the star bam file.
By the way, how did you generate the slurm output file from running a recipie?

By the way, we need to index the bam file also! I forgot about that =P

//JEsper

### CODE REVIEW

File: lib/MIP/Recipes/Analysis/Star_aln.pm [hunk 1]
```diff
@@ -168,6 +168,8 @@ sub analysis_star_aln {
     use MIP::Get::Parameter qw{ get_module_parameters };
     use MIP::IO::Files qw{ migrate_file };
     use MIP::Program::Alignment::Star qw{ star_aln };
+    use MIP::Program::Alignment::Picardtools
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Star_aln.pm]: Order AB

File: lib/MIP/Recipes/Analysis/Star_aln.pm [hunk 2]
```diff
@@ -320,8 +320,23 @@ sub analysis_star_aln {
...
                 two_pass_mode       => $active_parameter_href->{two_pass_mode},
-            }
+            },
+        );
+        say {$FILEHANDLE} $NEWLINE;
+
+        picardtools_addorreplacereadgroups(
+            {
+                FILEHANDLE              => $FILEHANDLE,
+                infile_path             => $outfile_path_prefix . ".bam",
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Star_aln.pm]: ".bam" -> $outfile_suffix 
I think this is ".bam" right?

File: lib/MIP/Recipes/Analysis/Star_aln.pm [hunk 3]
```diff
@@ -320,8 +320,23 @@ sub analysis_star_aln {
...
+                outfile_path            => $outfile_path_prefix . ".RG.bam",
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Star_aln.pm]: ".RG.bam" -> $DOT . q{RG} . $outfile_suffix
Do not forget to run perl critic on the module

**Comment by J35P312** [comment]
Hello there!
Now I have fixed it!

I still get some funky warning from perl critic:
Useless interpolation of literal string at line 332, column 28.  See page 51 of PBP.  (Severity: 1)

I skipped the bam index creation also, I see you do not create the bam index in your bwa_mem module, so I try to keep it similar =P.

**Comment by henrikstranneheim** [comment]
@J35P312 Did you pusgh your latest changes? regarding the warning:
Don't use double-quotes or qq// if your string doesn't require interpolation. This saves the interpreter a bit of work and it lets the reader know that you really did intend the string to be literal.

**Comment by J35P312** [comment]
Woopsie daisy!
I forgot to push my latest changes =P. But here they are!
I also removed the double quotes that perlcritic disliked

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**